src/domain/core/src/services/dataset_registry.rs

```diff
@@ -19,6 +19,7 @@ use crate::{DatasetHandleStream, GetDatasetError, ResolvedDataset};
 pub trait DatasetRegistry: Send + Sync {
     fn all_dataset_handles(&self) -> DatasetHandleStream<'_>;
 
+    // TODO: Private Datasets: replace AccountName with AccountID?
```

Comment:
We could be far from this at the moment. This would require getting rid of `alias` files first.